src/metabase/api/public.clj

```diff
@@ -303,6 +305,34 @@
      :html    (embed/iframe url width height)}))
 
 
+;;; ----------------------------------------------- Public Action ------------------------------------------------
+
+(defn- remove-action-non-public-columns
+  "Remove everyting from public `card` that shouldn't be visible to the general public."
+  [action]
+  (mi/instance
+   Action
+   (select-keys action [:description
+                        :database_id
+                        :name
+                        :type
+                        :dataset_query
+                        :model_id
+                        :id
+                        :parameter_mappings
+                        :visualization_settings
+                        :parameters])))
```

Comment:
@kulyk Can you check these columns are sufficient for rendering the public action page? This is the list of keys that are returned with the `GET "/action/:uuid"` response body.

Reply:
Yeah, even more than sufficient. I think we should definitely exclude `dataset_query` from here though, so we don't expose the database schema to everyone out there. Think we should only keep: `id` (UUID), `name`, `parameters`, and `visualization_settings`. The rest seem redundant

Reply:
Nice, narrowed it down here https://github.com/metabase/metabase/pull/27781/commits/b33a12f85924d0632f224007ad0adc24e035d55a